pythia/tasks/vqa/visual_genome/dataset.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import json
+import torch
+
+from pythia.tasks.vqa.vqa2 import VQA2Dataset
+from pythia.tasks.scene_graph_database import SceneGraphDatabase
+from pythia.common.sample import Sample, SampleList
+
+_CONSTANTS = {
+    "image_id_key": "image_id"
+}
+
+
+class VisualGenomeDataset(VQA2Dataset):
+    def __init__(self, dataset_type, imdb_file_index, config, *args, **kwargs):
+        super().__init__(dataset_type, imdb_file_index, config, *args, **kwargs)
+        self._name = "visual_genome"
+
+        self._return_scene_graph = config.return_scene_graph
+        self._return_objects = config.return_objects
+        self._return_relationships = config.return_relationships
+        self.scene_graph_db = None
+
+        should_build_scene_graph_db = (
+            self._return_scene_graph or self._return_objects
+            or self._return_relationships
+        )
+
+        if should_build_scene_graph_db:
+            scene_graph_file = config.scene_graph_files[dataset_type][imdb_file_index]
+            scene_graph_file = self._get_absolute_path(scene_graph_file)
+            self.scene_graph_db = SceneGraphDatabase(scene_graph_file)
+
+    def load_item(self, idx):
+        sample_info = self.imdb[idx]
+        sample_info = self._preprocess_answer(sample_info)
+        current_sample = Sample()
+
+        text_processor_argument = {"text": sample_info["question"]}
+
+        processed_question = self.text_processor(text_processor_argument)
+
+        current_sample.text = processed_question["text"]
```

Comment:
Are attributes like `text`, `question_id` created already or is this the first time they are created?

Reply:
First time. It is basically `__setattr__`.